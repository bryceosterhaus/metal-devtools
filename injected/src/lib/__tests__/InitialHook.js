jest.unmock('../InitialHook');

import InitialHook from '../InitialHook';

describe('InitialHook', () => {
	test('should initialize', () => {
		const Hook = new InitialHook();

		expect(Hook).toBeTruthy();
	});

	test('should add to components array', () => {
		const Hook = new InitialHook();

		Hook.add('foo');

		expect(Hook._roots).toMatchSnapshot();
	});

	test('should add to components array', () => {
		const Hook = new InitialHook();

		Hook.add('foo');
		Hook.add('bar');
		Hook.add('baz');

		expect(Hook.getAll()).toMatchSnapshot();
		expect(Hook.getAll()).toMatchSnapshot();
	});

	test('should check if components exist', () => {
		const Hook = new InitialHook();

		expect(Hook.hasRoots()).toBeFalsy();

		Hook.add('foo');

		expect(Hook.hasRoots()).toBeTruthy();
	});
});
