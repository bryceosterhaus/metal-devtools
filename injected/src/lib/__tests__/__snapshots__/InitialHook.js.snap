exports[`InitialHook should add to components array 1`] = `
Array [
  "foo",
]
`;

exports[`InitialHook should add to components array 2`] = `
Array [
  "foo",
  "bar",
  "baz",
]
`;

exports[`InitialHook should add to components array 3`] = `
Array [
  "foo",
  "bar",
  "baz",
]
`;
